keras_cv/models/convnext.py

```diff
@@ -0,0 +1,494 @@
+        x = layers.LayerNormalization(epsilon=1e-6, name=name + "_head_layernorm")(x)
+        x = layers.Dense(num_classes, activation=activation, name=name + "_head_dense")(
+            x
+        )
+        return x
+
+    return apply
+
+
+def ConvNeXt(
+    include_rescaling,
+    depths,
+    projection_dims,
+    drop_path_rate=0.0,
+    layer_scale_init_value=1e-6,
+    include_top=True,
+    weights=None,
+    input_shape=(None, None, 3),
+    input_tensor=None,
+    pooling=None,
+    classes=1000,
+    classifier_activation="softmax",
+    model_name="convnext",
+):
+    """Instantiates ConvNeXt architecture given specific configuration.
+
+    Args:
+      include_rescaling: whether or not to Rescale the inputs. If set to True,
+        inputs will be passed through a `Rescaling(1/255.0)` layer.
+      depths: An iterable containing depths for each individual stages.
+      projection_dims: An iterable containing output number of channels of
+      each individual stages.
+      drop_path_rate: Stochastic depth probability. If 0.0, then stochastic
+        depth won't be used.
+      layer_scale_init_value: Layer scale coefficient. If 0.0, layer scaling
+        won't be used.
+      include_top: Boolean denoting whether to include classification head to
+        the model.
+      weights: one of `None` (random initialization), `"imagenet"` (pre-training
+        on ImageNet-1k), or the path to the weights file to be loaded.
+      input_shape: optional shape tuple, defaults to (None, None, 3).
+      input_tensor: optional Keras tensor (i.e. output of `layers.Input()`).
+      pooling: optional pooling mode for feature extraction when `include_top`
+        is `False`.
+        - `None` means that the output of the model will be the 4D tensor output
+          of the last convolutional layer.
+        - `avg` means that global average pooling will be applied to the output
+          of the last convolutional layer, and thus the output of the model will
+          be a 2D tensor.
+        - `max` means that global max pooling will be applied.
+      classes: optional number of classes to classify images into, only to be
+        specified if `include_top` is True, and if no `weights` argument is
+        specified.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+      model_name: An optional name for the model.
+
+    Returns:
+      A `keras.Model` instance.
+
+    Raises:
+        ValueError: in case of invalid argument for `weights`,
+          or invalid input shape.
+        ValueError: if `classifier_activation` is not `softmax`, or `None`
+          when using a pretrained top layer.
+        ValueError: if `include_top` is True but `num_classes` is not 1000
+          when using ImageNet.
```

Comment:
Instead of specifying 1000/imagenet, let's just say if `num_classes` is not specified.